ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -129,49 +117,172 @@ def get_agent_step_result(self, agent_id: AgentId) -> StepResult:
             agent_mask = []
             for mask in self.action_mask:
                 agent_mask.append(mask[agent_index])
-        return StepResult(
+        return DecisionStep(
             obs=agent_obs,
             reward=self.reward[agent_index],
-            done=self.done[agent_index],
-            max_step=self.max_step[agent_index],
             agent_id=agent_id,
             action_mask=agent_mask,
         )
 
+    def __iter__(self) -> AgentId:
+        for agent_id in self.agent_id:
+            yield agent_id
+
+    def keys(self) -> List[AgentId]:
```

Comment:
Don't implement these. Forming the list is inefficient, and the default implementations will work in terms of `__iter__` and `__getitem__`